linode/manager
# refactor: [M3-8232] - Query Key Factory for Firewalls

## Description 📝

- Migrates Firewalls to use a query key factory 🏭 
- Migrates to single object notation to prepare for TanStack Query v5 (see [here](https://tanstack.com/query/latest/docs/framework/react/guides/migrating-to-v5#supports-a-single-signature-one-object)) 🔧 
- Performs some clean up and optimizations where possible 🧹 

## Preview 📷

> [!note]
> No UI changes expected

## How to test 🧪

- Test general firewall actions
- Test firewall actions on Linode Details page
  - We want to verify cache updates happen as expected
- Test firewall actions on NodeBalancer settings page
  - We want to verify cache updates happen as expected
- Verify all automated testing for Firewalls passes ✅ 

## As an Author I have considered 🤔
